pkg/rbd/rbd_util.go

```diff
@@ -205,9 +224,430 @@ func rbdStatus(pOpts *rbdVolume, userID string, credentials map[string]string) (
+	}
+	rbdVol.VolID, err = vi.ComposeVolID()
+	if err != nil {
+		return false, err
+	}
+
+	klog.V(4).Infof("Found existng volume (%s) with image name (%s) for request (%s)",
+		rbdVol.VolID, rbdVol.VolName, rbdVol.RequestName)
+
+	return true, nil
+}
+
+/*
+unreserveRBDSnap and unreserveRBDVol remove omaps associated with the snapshot and the image name,
+and also remove the corresponding request name key in the snaps or volumes omaps respectively.
+
+This is performed within the request name lock, to ensure that requests with the same name do not
+manipulate the omap entries concurrently.
+*/
+func unreserveRBDSnap(rbdSnap *rbdSnapshot, credentials map[string]string) error {
+	key, err := getRBDKey(rbdSnap.ClusterID, rbdSnap.AdminID, credentials)
+	if err != nil {
+		return err
+	}
+
+	// delete snap image omap (first, inverse of create order)
+	err = util.RemoveOMap(rbdSnap.Monitors, rbdSnap.AdminID, key, rbdSnap.Pool, snapImgOMap+rbdSnap.SnapName)
+	if err != nil {
+		if _, ok := err.(util.ErrOMapNotFound); !ok {
+			klog.V(4).Infof("failed removing oMap %s (%s)", snapImgOMap+rbdSnap.SnapName, err)
+			return err
+		}
+	}
+
+	// delete the request name omap key (last, inverse of create order)
+	err = util.RemoveOMapKey(rbdSnap.Monitors, rbdSnap.AdminID, key, rbdSnap.Pool,
+		snapsOMap, snapsOMapNameKey+rbdSnap.RequestName)
+	if err != nil {
+		klog.V(4).Infof("failed removing oMap key %s (%s)", snapsOMapNameKey+rbdSnap.RequestName, err)
+		return err
+	}
+
+	return nil
+}
+
+func unreserveRBDVol(rbdVol *rbdVolume, credentials map[string]string) error {
+	key, err := getRBDKey(rbdVol.ClusterID, rbdVol.AdminID, credentials)
+	if err != nil {
+		return err
+	}
+
+	// delete image omap (first, inverse of create order)
+	err = util.RemoveOMap(rbdVol.Monitors, rbdVol.AdminID, key, rbdVol.Pool, imgOMap+rbdVol.VolName)
+	if err != nil {
+		if _, ok := err.(util.ErrOMapNotFound); !ok {
+			klog.V(4).Infof("failed removing oMap %s (%s)", imgOMap+rbdVol.RequestName, err)
+			return err
+		}
+	}
+
+	// delete the request name omap key (last, inverse of create order)
+	err = util.RemoveOMapKey(rbdVol.Monitors, rbdVol.AdminID, key, rbdVol.Pool,
+		volOMap, volOMapNameKey+rbdVol.RequestName)
+	if err != nil {
+		klog.V(4).Infof("failed removing oMap key %s (%s)", volOMapNameKey+rbdVol.RequestName, err)
+		return err
+	}
+
+	return nil
+}
+
+/*
+reserverRBDSnap and reserveRBDVol add respective entries to the volumes and snapshots omaps, post
+generating a target snapshot or image name for use. Further, these functions create the snapshot or
+image name omaps, to store back pointers to the CSI generated request names.
+
+This is performed within the request name lock, to ensure that requests with the same name do not
+manipulate the omap entries concurrently.
+*/
+func reserveRBDSnap(rbdSnap *rbdSnapshot, credentials map[string]string) error {
+	var vi util.VolumeIdentifier
+
+	// generate a uuid for the snap image name
+	uuid := uuid.NewUUID().String()
```

Comment:
can we name this variable `uuid` to something else, because package name imported is also `uuid`

Reply:
Will address in a future update to the PR.